src/contracts/GPv2Settlement.sol

```diff
@@ -279,8 +295,27 @@ contract GPv2Settlement {
     function orderUid(bytes32 orderDigest, address owner)
         private
         pure
-        returns (bytes32)
+        returns (bytes32 uid)
     {
-        return keccak256(abi.encodePacked(orderDigest, owner));
+        // NOTE: Use the 64 bytes of scratch space starting at memory address 0
+        // for computing this hash instead of allocating. We hash a total of 52
+        // bytes and write to memory in **reverse order** as memory operations
+        // write 32-bytes at a time and we want to use a packed encoding:
+        //
+        //       |           111111111122222222223333333333444444444455
+        // byte  | 0123456789012345678901234567890123456789012345678901
+        // ------+-----------------------------------------------------
+        // field | [..........orderDigest.........][.......owner......]
+        // ------+-----------------------------------------------------
+        // write |                     [.............owner............]
+        //       | [..........orderDigest.........]
+        //
+        // <https://docs.soliditylang.org/en/v0.7.5/internals/layout_in_memory.html>
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            mstore(20, owner)
```

Comment:
Maybe comment specifically that bytes 20-32 will be 0x (took me a while to understand this from the larger text blob above).